synapse/handlers/devicemessage.py

```diff
@@ -44,13 +45,32 @@ def __init__(self, hs: "HomeServer"):
         self.store = hs.get_datastore()
         self.notifier = hs.get_notifier()
         self.is_mine = hs.is_mine
-        self.federation = hs.get_federation_sender()
 
-        hs.get_federation_registry().register_edu_handler(
-            "m.direct_to_device", self.on_direct_to_device_edu
-        )
+        # We only need to poke the federation sender explicitly if its on the
```

Comment:
```suggestion
        # We only need to poke the federation sender explicitly if it's on the
```

Reply:
... how does it work if it's on a different instance?

Reply:
I've expanded the comment to:

>We only need to poke the federation sender explicitly if its on the
same instance. Other federation sender instances will get notified by
the `generic_worker` when it sees it in the to-device replication
stream.